### PR TITLE
Kfairise/add test to install both agent

### PR DESCRIPTION
**Pull request by KevinFairise2**
Add more tests to try to install dd agent together with OPW agent

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -160,6 +160,8 @@ test-apm-injection:
 
 test-observability-pipelines-worker:
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64:$DATADOG_AGENT_BUILDIMAGES
+  rules:
+  - if: '$CI_PIPELINE_SOURCE == "push"'
```
Comment (spencergilbert) [inline .gitlab-ci.yml:164]: **❓ question:** ‏ Was this for testing or intended to be left in? If intentional, why is it the only job with this rule?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -172,8 +174,21 @@ test-observability-pipelines-worker:
       - IMAGE: rockylinux:9.0
       - IMAGE: ubuntu:20.04
       - IMAGE: ubuntu:22.04
+      - IMAGE: amazonlinux:2
```
Comment (spencergilbert) [inline .gitlab-ci.yml:177]: **💬 suggestion:** ‏ Missed the additional variable in this matrix variant.
```suggestion
      - IMAGE: amazonlinux:2
        INSTALL_CLASSIC_AGENT: --opw-install-classic-agent install_script_agent7.sh
```

File: test/op-worker-test.sh [hunk 3]
```diff
@@ -1,7 +1,12 @@
 #!/bin/bash -e
 
+if [[ -v "$DD_OPW_INSTALL_CLASSIC_AGENT" ]]; then
+$DD_OPW_INSTALL_CLASSIC_AGENT
+fi
+
 $SCRIPT
 
+
```
Comment (spencergilbert) [inline test/op-worker-test.sh:9]: **🥜 nitpick:** ‏ Stray linebreak added?
```suggestion
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (KevinFairise2) [reply .gitlab-ci.yml:164]: It was intended to be here, so this job is not triggered when this pipeline is triggered from `datadog-agent` repository.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**